meson.build

```diff
@@ -0,0 +1,36 @@
+project(
+    'ddbus',
+    'd',
+    meson_version: '>=0.50',
+    version: '3.0.0',
+)
+
+project_soversion = 0
+
+d_mod_version = []
+
+src_inc = include_directories('source')
+
+prefix = get_option('prefix')
+include_dir = join_paths(prefix, get_option('includedir'), 'd', 'apk-polkit')
+
+vcs_tag = run_command('git', 'rev-parse', '--short', 'HEAD').stdout().strip()
+if vcs_tag == ''
+    corecollector_version = meson.project_version()
+else
+    corecollector_version = meson.project_version() + '-@0@'.format (vcs_tag)
+endif
+
+compiler_id = meson.get_compiler('d').get_id()
+
+link_runtime_shared_arg = []
+
+if compiler_id == 'llvm'
+    link_runtime_shared_arg = ['-link-defaultlib-shared']
+elif compiler_id == 'dmd'
+    link_runtime_shared_arg = ['-defaultlib=phobos2', '-debuglib=phobos2']
+elif compiler_id == 'gcc'
+    link_runtime_shared_arg = ['-shared-libphobos']
+endif
```

Comment:
Is this really needed? It seems wrong... Something like this should be fixed in the meson upstream.

Reply:
It should be, yes, but this is a workaround for the time being.  I'll see if I get to fixing this in Meson soon.

Reply:
I just commented it out locally and tried it and I don't run into any issue. Which meson version are you using?

Reply:
The problem is that this way meson will link a static version of your stdlib into this lib, which isn't what you want. If you were to link the shared lib containing the static stdlib against something else that either uses a different version of the stdlib (shared) or any version (static) it'll blow up because the DRuntime can't be initialized.

Reply:
No. It is working fine here.

```
source/ddbus/libddbus.so (interpreter => None)
    libphobos2-ldc-shared.so.89 => /usr/lib/libphobos2-ldc-shared.so.89
    libdruntime-ldc-shared.so.89 => /usr/lib/libdruntime-ldc-shared.so.89
    librt.so.1 => /usr/lib/librt.so.1
    libdl.so.2 => /usr/lib/libdl.so.2
    libpthread.so.0 => /usr/lib/libpthread.so.0
    libm.so.6 => /usr/lib/libm.so.6
    libgcc_s.so.1 => /usr/lib/libgcc_s.so.1
    libc.so.6 => /usr/lib/libc.so.6
    ld-linux-x86-64.so.2 => /usr/lib/ld-linux-x86-64.so.2
```

Post your `/etc/ldc2.conf`.